pkg/crypto/crypto_test.go

```diff
@@ -111,3 +112,85 @@ func decryptConfigMap(configMap string, keyPath string, noncePath string) string
 
 	return string(plaintextBytes)
 }
+
+
+func TestDecryptConfigMapWrongkey(t *testing.T)  {
+
+        /*Test Objective : Negative test: In Symmetric encryption only one key (a secret key) is used to
+         both encrypt and decrypt. If different key is passed for decrypt, decrypt function should fail.
+         In this test, wrong key value is being used for decryption, test should FAIL*/
+
+        testString := "Sample String to Test Symmetric Encryption"
+
+        dir1, err := ioutil.TempDir("", "TestEncryption-2")
+        if err != nil {
+                t.Error(err)
+        }
+        defer os.RemoveAll(dir1)
+
+        keyPath := dir1 + "symmKeyFile"
+        wrongkeyPath := dir1 + "wrongsymmKeyFile"
+
+        key := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, key); err != nil {
+                t.Error(err)
+        }
+
+        wrongkey := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, wrongkey); err != nil {
+                t.Error(err)
+        }
+
+
+        //Encode the Key
+        keyEnc := b64.StdEncoding.EncodeToString(key)
+        if err != nil {
+                t.Error(err)
+        }
+
+        //encode wrong key
+        wrongkeyEnc := b64.StdEncoding.EncodeToString(wrongkey)
+        if err != nil {
+                t.Error(err)
+        }
+
+
+        err = ioutil.WriteFile(keyPath, []byte(keyEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+        err=ioutil.WriteFile(wrongkeyPath, []byte(wrongkeyEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+
+
+        noncePath := dir1 + "nonceFile"
+        nonce := make([]byte, 12)
+        if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+                t.Error(err)
+        }
+
+        //Encode the nonce
+        nonceEnc := b64.StdEncoding.EncodeToString(nonce)
+
+        err = ioutil.WriteFile(noncePath, []byte(nonceEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+
+        encConfigMap, err := EncryptConfigMap([]byte(testString), keyPath, noncePath)
+
+        if err != nil {
+                t.Errorf("Failed to encrypt configMap %v", err)
+        }
+
+        wrongdecryptedConfigMap := decryptConfigMap(encConfigMap, wrongkeyPath, noncePath)
+        fmt.Printf("What is being returned by decrypt function on getting a wrong key = %s", wrongdecryptedConfigMap)
+
+        if testString != wrongdecryptedConfigMap {
```

Comment:
If you use the incorrect key to decrypt the configmap, won't this condition be always true? Also, `decryptConfigMap` should return `err` instead of panicking (https://github.com/IBM/raksh/blob/6b2e4df4e38bd747e098cd8c0f8119c037cde1bc/pkg/crypto/crypto_test.go#L90). That will make it easier to handle it here.

Reply:
I agree. That's exactly what I had in mind, when I wanted to write negative scenarios. As it was something pre-existing, I did not change it. However, I will discuss this with my tech lead and update further on this.

Reply:
Thanks for the suggestions. Do you prefer I address this in this PR or create a separate PR for this

Reply:
Please suggest if it would be fine to replace all panic calls with err calls in decrypt function? 
FYI, @harche @bpradipt @mkumatag

Reply:
Let's deal with this once we have the clarity on why this change is needed.

Reply:
yes